maven-xml-impl/src/main/java/org/apache/maven/internal/xml/XmlNodeWriter.java

```diff
@@ -18,49 +18,146 @@
  */
 package org.apache.maven.internal.xml;
 
-import java.io.PrintWriter;
+import javax.xml.stream.XMLOutputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamWriter;
+
 import java.io.Writer;
 import java.util.Map;
 
 import org.apache.maven.api.xml.XmlNode;
-import org.codehaus.plexus.util.xml.PrettyPrintXMLWriter;
-import org.codehaus.plexus.util.xml.XMLWriter;
+import org.codehaus.stax2.util.StreamWriterDelegate;
 
 /**
  *
  */
 public class XmlNodeWriter {
     public static void write(Writer writer, XmlNode dom) {
-        write(new PrettyPrintXMLWriter(writer), dom);
-    }
-
-    public static void write(PrintWriter writer, XmlNode dom) {
-        write(new PrettyPrintXMLWriter(writer), dom);
-    }
-
-    public static void write(XMLWriter xmlWriter, XmlNode dom) {
-        write(xmlWriter, dom, true);
+        try {
+            XMLOutputFactory factory = new com.ctc.wstx.stax.WstxOutputFactory();
+            factory.setProperty(XMLOutputFactory.IS_REPAIRING_NAMESPACES, false);
+            factory.setProperty(com.ctc.wstx.api.WstxOutputProperties.P_USE_DOUBLE_QUOTES_IN_XML_DECL, true);
+            factory.setProperty(com.ctc.wstx.api.WstxOutputProperties.P_ADD_SPACE_AFTER_EMPTY_ELEM, true);
+            XMLStreamWriter serializer = new IndentingXMLStreamWriter(factory.createXMLStreamWriter(writer));
+            write(serializer, dom);
+            serializer.close();
+        } catch (XMLStreamException e) {
+            throw new RuntimeException("Unexpected error when writing XML", e);
```

Comment:
This should not be a runtime exception. XMLStreamException is better

Reply:
No problem, that method is used in `XmlNodeImpl#toString()`, so that will just move the catch block elsewhere...

Reply:
It's a public method. It's not safe to assume it's only used by toString.
Also toString should not throw any exceptions, runtime or otherwise. If an exception can be thrown, it should be handled in a way that still returns a string.